test/distributed/test_c10d_nccl.py

```diff
@@ -1937,7 +1937,8 @@ def first_bucket_size(ddp_bucket_mb):
                         opt = torch.optim.SGD(m.parameters(), lr=0.1)
                         opt_ddp = torch.optim.SGD(m_ddp.parameters(), lr=0.1)
                         has_half = any(p.dtype is torch.half for p in m.parameters())
-                        tol = 1.0e-3 if has_half else 1.0e-5
+                        atol = 1.0e-3 if has_half else 1.0e-4
+                        rtol = 1.0e-3
```

Comment:
Hmm, reduction over registered buffer should lead to better precision if using NVLS, and at least should not be worse than the non-registered baseline if using ring. Do you know why this needs to be relaxed?

Reply:
In general yes, but I am not 100% sure about a two-rank case (if that's the world size of this test)

Reply:
Hmm, IIUC, with rank=2, floating point addition shouldn't suffer from associativity problem at all. Which means we should expect identical results?